### PR TITLE
Upgrade various async dependencies

### PR DESCRIPTION
**Pull request by mjgarton**
This upgrades tokio, futures & mysql_async to recent versions.  Doing so
involved reworking the async code in the tests to use the modern rust
async/await style, and updating code to work with the more recent
method names in mysql_async.

### CODE REVIEW

File: tests/async.rs [hunk 1]
```diff
@@ -768,7 +757,7 @@ fn prepared_nulls() {
             assert!(!params[1].value.is_null());
             assert_eq!(
                 params[0].coltype,
-                myc::constants::ColumnType::MYSQL_TYPE_SHORT
+                myc::constants::ColumnType::MYSQL_TYPE_NULL
```
Comment (jonhoo) [inline tests/async.rs:760]: Weird. It's wild that this _used_ to work.

**Review by jonhoo — CHANGES_REQUESTED** [review]
Excellent, thank you! A few minor notes.

File: tests/async.rs (hunk 1 above)
Comment (mjgarton) [reply tests/async.rs:760]: I found this surprising too, but it looks correct now and I didn't dig much into why it worked earlier.

**Review by jonhoo — APPROVED** [review]
Thank you!

**Pull request merged**